### ISSUE TITLE
Value autocomplete wrong behavior

### ISSUE THREAD

**Issue by Bibi56** [openstreetmap/iD#6441]
Entering stan for the key camp_site proposes standard (correct) but also standart (wrong).
For German-speaking people the second may sound good, it is not.
Where the possible values are limited, only valid values (present on the Wiki, accepted proposals...) should be proposed.
Elsewhere, ill formed values (like my-proposed-feature where my_proposed_feature is proposed) should be eliminated.

**Comment by BjornRasmussen** [comment]
iD currently uses taginfo, which stores all commonly used keys.  This means that if a key has a lot of use, the key will be suggested to the user typing in the raw tag editor, regardless of whether or not the key is approved

**Comment by Bibi56** [comment]
I'm speaking about values, not keys.
I check the occurrence of standard (82) and standart (2).
2 (two) usages worldwide is not what I would call "a lot of use".
And, even without checking with wiki recommended values, the difference between the 2 usages (with similar values) is huge.

**Comment by bhousel** [comment]
Thanks for checking @Bibi56  - if there are just 2 instances worldwide and it's an obvious mistake, just change them.

**Comment by Bibi56** [comment]
That's exactly my point: because they are almost not used, they shouldn't get proposed!
BTW, I changed the two invalid values.

**Comment by Marc-marc-marc** [comment]
2 suggestions to improve the quality of proposals made through taginfo:
- hide values that have less than x% usage without a wiki page, this should hide typing errors
- hide depreciated values

**Comment by BjornRasmussen** [comment]
That could turn into controversy about which tags are considered to be depreciated and which tags aren't.  They best solution is to just remove depreciated values from the database so iD doesn't suggest them.

**Comment by quincylvania** [comment]
iD does hide values from the suggestion that are on the deprecated list (#6084).

I actually don't mind low-usage tags getting suggested. Maybe someone is mapping with a great new tag they created. More importantly, there are plenty of "accepted" tags with relatively low usage. Some people might be confused or angry if these didn't appear as expected.

**Comment by Marc-marc-marc** [comment]
> That could turn into controversy about which tags are considered to be depreciated

this would be surprising since the wiki page listing the deprecated tags is seen by a large number of contributors.

>  just remove depreciated values from the database

it's obviously always better/&needed to fix the database but it's a misunderstanding of osm to believe that all deprecated tags can quickly disappear.
moreover, the list of deprecated tags does not prevent typos.
today if someone makes a typo, iD will propose the typo in spite of itself since it exists in taginfo.
that's why I proposed a simple way to improve the quality of iD suggestions while avoiding that iD have to create a black list by hand, which would be an endless job

**Comment by Bibi56** [comment]
For low usage values like standart, a soundex comparison with common values could help

**Comment by Bibi56** [comment]
Yesterday a contributor used `dispersed` as value because it was proposed in the drop-down list for camp_site. At this time nobody was using `camp_site=dispersed`. So some values must come from outside of taginfo.

**Comment by quincylvania** [comment]
> Yesterday a contributor used dispersed as value because it was proposed in the drop-down list for camp_site.

@Bibi56 How can you tell? iD doesn't suggest this for me (though maybe someone removed the usage).

![Screen Shot 2019-06-10 at 8 34 19 AM](https://user-images.githubusercontent.com/2046746/59195707-a39fff00-8b5a-11e9-94f6-f190c09f3be7.png)

>  At this time nobody was using camp_site=dispersed. So some values must come from outside of taginfo.

Nope, the values in the "All tags" editor come only from taginfo. [Check the code](https://github.com/openstreetmap/iD/blob/16443e2284ae2eff5f5be5e9e51292d2ec9641f5/modules/ui/raw_tag_editor.js#L389-L399). Some fields in the "All fields" section do have [predefined suggestions](https://github.com/openstreetmap/iD/tree/master/data/presets#strings), but there isn't a field for `camp_site`.

**Comment by Bibi56** [comment]
@quincylvania Yes the creator of the `node camp_site=dispersed` removed the tag.
I can tell because I wrote him "Hi, I've seen you are using non standard values.
dispersed is not a possible value for camp_site. basic is probably the correct value here.". At the time he added the value, taginfo was unaware of this value. One day later it gave the value (that's how I found this usage). The user answered me "You are correct about no “dispersed” camp_site, nothing in the Wiki, but then why does it come up as possible value when you scroll down in values?" and for the changeset he was using ID. Therefore my deduction. Today I don't see "dispersed" as proposed value either which confirms what you're saying.
